wit/bindgen/generator.go

```diff
@@ -1800,31 +1800,33 @@ func (g *generator) defineImportedFunction(_ wit.Ident, f *wit.Function, decl fu
 	b.WriteString("}\n\n")
 
 	// Emit wasmimport function
-	stringio.Write(&b, "//go:wasmimport ", decl.linkerName, "\n")
-	b.WriteString("//go:noescape\n")
-	b.WriteString("func ")
+	externFile := g.externFileFor(owner)
+
+	stringio.Write(externFile, "//go:wasmimport ", decl.linkerName, "\n")
+	externFile.WriteString("//go:noescape\n")
+	externFile.WriteString("func ")
 	if decl.wasm.isMethod() {
-		stringio.Write(&b, "(", decl.wasm.receiver.name, " ", g.typeRep(file, decl.wasm.receiver.dir, decl.wasm.receiver.typ), ") ", decl.wasm.name)
+		stringio.Write(externFile, "(", decl.wasm.receiver.name, " ", g.typeRep(externFile, decl.wasm.receiver.dir, decl.wasm.receiver.typ), ") ", decl.wasm.name)
 	} else {
-		b.WriteString(decl.wasm.name)
+		externFile.WriteString(decl.wasm.name)
 	}
-	b.WriteString(g.functionSignature(file, decl.wasm))
+	externFile.WriteString(g.functionSignature(externFile, decl.wasm))
 
-	b.WriteString("\n\n")
+	externFile.WriteString("\n\n")
 
 	// Emit shared types
 	if t, ok := compoundParams.typ.(*wit.TypeDef); ok {
 		td, _ := g.typeDecl(dir, t)
-		stringio.Write(&b, "// ", td.name, " represents the flattened function params for [", decl.wasm.name, "].\n")
-		stringio.Write(&b, "// See the Canonical ABI flattening rules for more information.\n")
-		stringio.Write(&b, "type ", td.name, " ", g.typeDefRep(file, dir, t, td.name), "\n\n")
+		stringio.Write(externFile, "// ", td.name, " represents the flattened function params for [", decl.wasm.name, "].\n")
+		stringio.Write(externFile, "// See the Canonical ABI flattening rules for more information.\n")
+		stringio.Write(externFile, "type ", td.name, " ", g.typeDefRep(externFile, dir, t, td.name), "\n\n")
 	}
 
 	if t, ok := compoundResults.typ.(*wit.TypeDef); ok {
 		td, _ := g.typeDecl(dir, t)
-		stringio.Write(&b, "// ", td.name, " represents the flattened function results for [", decl.wasm.name, "].\n")
-		stringio.Write(&b, "// See the Canonical ABI flattening rules for more information.\n")
-		stringio.Write(&b, "type ", td.name, " ", g.typeDefRep(file, dir, t, td.name), "\n\n")
+		stringio.Write(externFile, "// ", td.name, " represents the flattened function results for [", decl.wasm.name, "].\n")
+		stringio.Write(externFile, "// See the Canonical ABI flattening rules for more information.\n")
+		stringio.Write(externFile, "type ", td.name, " ", g.typeDefRep(externFile, dir, t, td.name), "\n\n")
```

Comment:
This type should be emitted in the main file, not the wasm file, so implementors don’t need to reimplement this type when compiling without `wasm` tag.